onnxruntime/contrib_ops/cuda/fused_conv.cc

```diff
@@ -100,6 +100,10 @@ class FusedConv : public onnxruntime::cuda::Conv<T, false> {
   Status MapMode(const std::string& activaton_mode) {
     if (activaton_mode == "Relu") {
       activation_mode_ = cudnnActivationMode_t::CUDNN_ACTIVATION_RELU;
+    } else if (activaton_mode == "Tanh") {
```

Comment:
Just fyi - I think adding new activation support here may also need corresponding logic update in the optimizer that generates the `FusedConv` node. Please see - https://github.com/microsoft/onnxruntime/blob/dd3b2cefd6abdd6a0fe6f6701d26d1299f87f1fa/onnxruntime/core/optimizer/conv_activation_fusion.cc#L192C7-L192C24.

I think the most common pattern is Conv -> Add -> Activation (There is almost always an `Add` after `Conv`, I believe ?) and we only support Conv -> Add -> Relu, please see here- https://github.com/microsoft/onnxruntime/blob/dd3b2cefd6abdd6a0fe6f6701d26d1299f87f1fa/onnxruntime/core/optimizer/conv_activation_fusion.cc#L227. So in order to truly leverage the extended activation support provided by CuDNN, we also need enhancements there if I understand correctly. 

It may be advisable to add a corresponding test for every enhancement/change in this kernel as it is an important optimization that is invoked in almost all image models.

Reply:
Yes I have altered those in another branch, but decided not to push those changes as I feel like exposing a compile capability on the CUDA EP will actually be the cleaner way to do this.